src/lint-changed.ts

```diff
@@ -162,33 +161,31 @@ export async function lintChanged() {
     return;
   }
 
-  let errors = 0;
-
-  Object.entries(lintConfig)
-    .map(([glob, cmds]) => [glob, Array.isArray(cmds) ? cmds : [cmds]])
-    .forEach(async ([glob, commands]) => {
-      µ(changedFiles, glob, {
-        dot: true,
-        matchBase: !glob.includes("/"),
-      }).forEach((file) => {
-        pEach(commands, (command) => {
-          log(`${command} ${file}`);
-          return limit(() =>
-            runCommand(`${command} ${file}`)
-              .then((o) =>
-                console.log(blue(dim(o)))
-              )
-              .catch((o) => {
-                console.log(red(o));
-                errors++;
-              })
-          );
-        })
-        .then(() => {
-          if (errors) {
-            process.exit(1);
-          }
-        });
-      });
+  const globAndCommands = Object.entries(lintConfig)
+    .map(([glob, cmds]) => [glob, Array.isArray(cmds) ? cmds : [cmds]]);
+
+  const results = await Promise.all(globAndCommands.map(async ([glob, commands]) => {
+    const files = µ(changedFiles, glob, {
+      dot: true,
+      matchBase: !glob.includes("/"),
     });
+
+    return await Promise.all(files.map((file) => limit(async () => {
+      // Stop running commands for current file on first command that fails
+      try {
+        for (const command of commands) {
+          const o = await runCommand(`${command} ${file}`);
+          console.log(blue(dim(o)));
+        }
+      } catch (e) {
+        error(e.message.trimLeft());
+        return e;
+      }
+    })));
+  }));
+
+  // In case there were any errors during execution, make sure to exit with error
+  if (results.some((command) => command.some((e) => e))) {
```

Comment:
maybe renaming `results` to `globResults` and `command` to `fileResults` will be easier to understand

```
globResults.some((fileResults) => fileResults.some((e) => e))
```